### PR TITLE
Fixed slat angle control and slat angle reporting issues for VB in equivalent layer window model

### PR DESCRIPTION
**Pull request by Nigusse**
Issue #5750

Pull request overview
---------------------
This is a fix for defects in Equivalent Layer Window Model Venetian Blind (VB) slat angle control and slat angle reporting issues. Now the slat angle control works as expected and the slat angle are reported for the three control modes. Also fixed the slat angle calculation for BlockBeamSolar control mode. Diffs are expected for BlockBeamSolar Slat Angle Control. Also swapped vertical and horizontal profile angles for equivalent layer window model.

### Work Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] At least one of the following appropriate labels must be added to this PR to be consumed into the changelog:
   - Defect: This pull request repairs a github defect issue.  The github issue should be referenced in the PR description
   - Performance: This pull request includes code changes that are directed at improving the runtime performance of EnergyPlus

### Review Checklist
This will not be exhaustively relevant to every PR.
 - [ ] Code style (parentheses padding, variable names)
 - [ ] Functional code review (it has to work!)
 - [ ] If defect, results of running current develop vs this branch should exhibit the fix
 - [ ] CI status: all green or justified
 - [ ] Performance: CI Linux results include performance check -- verify this
 - [ ] Unit Test(s)
 - C++ checks:
   - [ ] Argument types
   - [ ] If any virtual classes, ensure virtual destructor included, other things
 - IDD changes:
   - [ ] Verify naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs
 - [ ] Documentation changes in place
 - [ ] Changed docs build successfully
 - [ ] If output changes, including tabular output structure, add to output rules file for interfaces

### CODE REVIEW

**Comment by Nigusse** [comment]
The defect has been fixed and ready for review.

**Comment by Myoldmopar** [comment]
Yikes.  I tried resolving the conflict with the online editor.  Looks like it was trying to fix our whitespace for us, inserting spaces.  It looks like the conflict is just that two branches added some using statements.  No big deal.  I can fix that and push this back up.  I'll make sure it is updated with develop also.

File: src/EnergyPlus/WindowEquivalentLayer.cc [hunk 1]
```diff
@@ -8241,14 +8243,20 @@ namespace WindowEquivalentLayer {
...
 
 		// TODO handle vert blind cases etc
-		RAT = L.S * std::cos( OMEGA_DEG ) / L.W;
-		// limit upward slat angle to horiz = max visibility
-		VB_CriticalSlatAngle = max( 0.0, RadiansToDeg * std::asin( RAT ) - OMEGA_DEG );
+
+		// issue #5750, commented out because it does not block the solar beam
+		//RAT = L.S * std::cos( OMEGA_DEG ) / L.W;
+		//// limit upward slat angle to horiz = max visibility
+		//VB_CriticalSlatAngle = max( 0.0, RadiansToDeg * std::asin( RAT ) - OMEGA_DEG );
```
Comment (Myoldmopar) [inline src/EnergyPlus/WindowEquivalentLayer.cc]: Should just clean these out.  With our version control we can always look back and see where the file was modified, and with good commit messages we can see why.

File: src/EnergyPlus/WindowEquivalentLayer.cc [hunk 2]
```diff
@@ -9259,16 +9267,16 @@ namespace WindowEquivalentLayer {
 		ProfAngVer = 0.0;
 		ConstrNum = Surface( SurfNum ).Construction;
 		EQLNum = Construct( Surface( SurfNum ).Construction ).EQLConsPtr;
-
+
```
Comment (Myoldmopar) [inline src/EnergyPlus/WindowEquivalentLayer.cc]: Stray whitespace

File: idd/Energy+.idd.in [hunk 3]
```diff
@@ -6252,9 +6252,12 @@ WindowMaterial:Blind:EquivalentLayer,
...
        \type real
        \units deg
-       \minimum 0
-       \maximum 180
+       \minimum -90
+       \maximum 90
        \default 45
+       \note Slat angle is +ve if the tip of the slat front face is tilted upward, else  
+       \note the slat angle is -ve if the tip of the slat front face is tilted downward. 
+       \note The slat angle varies between -90 to +90. The default value is 45 degrees.
```
Comment (Myoldmopar) [inline idd/Energy+.idd.in:6260]: OK.  So this needs a transition rule to subtract 90 degrees from the original input?

**Review by Myoldmopar — CHANGES_REQUESTED** [review]
@Nigusse This looks pretty good.  This will need a transition rule to change the user's angle input right?  Or was it just being misinterpreted and we need to leave their input alone?

Other minor comments that will get cleaned up if transition rules or other code changes are needed.

File: src/EnergyPlus/WindowEquivalentLayer.cc (hunk 1 above)
Comment (Nigusse) [reply src/EnergyPlus/WindowEquivalentLayer.cc]: Agree. I will clean that.

File: idd/Energy+.idd.in (hunk 3 above)
Comment (Nigusse) [reply idd/Energy+.idd.in:6260]: Probably Slat Angle > 0 should remain the same but if greater than 90 then 90 minus Slat Angle should apply. I will double check this transition rule.

File: src/EnergyPlus/WindowEquivalentLayer.cc (hunk 1 above)
Comment (Nigusse) [reply src/EnergyPlus/WindowEquivalentLayer.cc]: Cleaned commented code lines.

File: src/EnergyPlus/WindowEquivalentLayer.cc (hunk 2 above)
Comment (Nigusse) [reply src/EnergyPlus/WindowEquivalentLayer.cc]: Removed white space.

File: idd/Energy+.idd.in (hunk 3 above)
Comment (Nigusse) [reply idd/Energy+.idd.in:6260]: Added transition rule (both *.md and *.f90 files).

Comment (Myoldmopar) [reply idd/Energy+.idd.in:6260]: Great.  I'll check it out once you get the changes pushed up.

Comment (Nigusse) [reply idd/Energy+.idd.in:6260]: I run into merge conflicts and had to commit twice.

File: src/Transition/CreateNewIDFUsingRulesV8_8_0.f90 [hunk 4]
```diff
@@ -803,6 +803,16 @@ SUBROUTINE CreateNewIDFUsingRules(EndOfFile,DiffOnly,InLfn,AskForInput,InputFile
                   ENDIF
                 ENDDO
 
+              CASE('WINDOWMATERIAL:BLIND:EQUIVALENTLAYER')
+                 ObjectName='WindowMaterial:Blind:EquivalentLayer'
+                 CALL GetNewObjectDefInIDD(ObjectName,NwNumArgs,NwAorN,NwReqFld,NwObjMinFlds,NwFldNames,NwFldDefaults,NwFldUnits)
+                 OutArgs(1:24) = InArgs(1:24)
+                 IF (inArgs(6) >= '90') THEN
```
Comment (Myoldmopar) [inline src/Transition/CreateNewIDFUsingRulesV8_8_0.f90]: Umm. Does this actually compile and run?

Comment (Nigusse) [reply src/Transition/CreateNewIDFUsingRulesV8_8_0.f90]: Nope. I never run transition codes.

Comment (Nigusse) [reply src/Transition/CreateNewIDFUsingRulesV8_8_0.f90]: I will push a corrected version.

**Comment by Myoldmopar** [comment]
@Nigusse Hey, thanks for going ahead and adding the transition rule.  I was happy to do it, but thanks for forging ahead on it.  

I am just wanting to verify the transition.  If before it went from 0-180, and now it goes from -90 to 90, why aren't you just shifting the value down by 90?  Maybe I don't fully grok the way in which the original code was broken, so I don't see why you are only transitioning this way.  If it is right, can you can clarify it one more time, and if all is good we'll merge this in.  (I'll test the transition locally now.)

**Comment by mjwitte** [comment]
Who added the equivalent layer model?  If we change the equivalent layer blinds to be -90 to +90 while regular blinds are 0 to 180 that seems a bit confusing.

**Comment by Nigusse** [comment]
@Myoldmopar @mjwitte The original equivalent window (ASHWAT) model uses the +90/-90 convention while EPlus uses 0 - 180 convention. So, the equivalent window blind model is based on +90/-90 convention but the idd was wrong (adopted Eplu's convention). @mjwitte I added the equivalent window model to Eplus but it was added in hasty situation when the original contractors did not deliver.

**Comment by Nigusse** [comment]
@Myoldmopar Just to clarify the slat angle convention in equivalent window model did not change only the idd changed to match the model in the code.  So, from functional point of view the model was already using +90/-90 convention.

**Comment by Myoldmopar** [comment]
OK, so do we really not have an example file for this object?  I find several `WindowMaterial:Blind`, but not `:EquivalentLayer`.  OK, well, I took the defect file, which was in 8.5, transitioned it up to 8.7 and then verified the transition works as expected.  If I use an angle < 90, it leaves it.  Anything over it and it subtracts 90.  So now the angle works as the code has expected it to work.  I think this is done.  Windows 32-bit is coming back with 2568 or 2568 tests passing, so that's about as good as I'd expect from CI.  I'll merge this.

**Review by Myoldmopar — APPROVED** [review]

**Pull request merged**